Automated docker build and push task in Travis CI

mcarmonaa opened the pull request:
I followed the same structure bblfsh use to push the server images.

This PR hasn't been tested because it needs to be merged and tagged on master.

For the moment the image tag is built as `dev-[7 first digits from spark-api HEAD commit]-dirty`

Docker credentials are already set in travis.

ajnavarro commented on Makefile:
```diff
@@ -0,0 +1,68 @@
+$(subst --,:,$(1))
+endef
+
+# if TRAVIS_TAG defined DOCKER_VERSION is overrided
```
s/DOCKER_VERSION/VERSION/

ajnavarro commented on .travis.yml:
```diff
@@ -73,4 +73,6 @@ script:
   - ./sbt assembly
```
you should change this to use make commands, to keep things consistent

ajnavarro commented on Makefile:
```diff
@@ -0,0 +1,68 @@
+# Rules
+all: clean build
+
+clean:
```
add test here too

ajnavarro commented:
@mcarmonaa Can you also add some documentatiion of how to start using the docker container please?

mcarmonaa commented:
Last modifications:

- Updated makefile to include some rules to use in travis, and some rules to build and run the jupyter server container
- Updated README with a new section `Playing around with spark-api on Jupyter`to use our container when it will be available on docker hub.
- Updated README section 'Development' with instructions about how to build and run a jupyter docker container that uses the last build of the project.

ajnavarro commented on Makefile:
```diff
@@ -0,0 +1,92 @@
+	./sbt test:scalastyle
+
+travis-cover:
+	./sbt ++$TRAVIS_SCALA_VERSION jacoco:cover
```
I think this is creating some problems on Travis build

mcarmonaa replied:
I removed this rule

ajnavarro commented on .travis.yml:
```diff
@@ -67,10 +67,12 @@ before_script:
-  - ./sbt test:scalastyle
+  - make scalastyle
+  - make test-scalastyle
   - ./sbt ++$TRAVIS_SCALA_VERSION jacoco:cover
```
ahaha, I think this is not the correct solution for that problem. You should change this to: ./sbt ++${TRAVIS_SCALA_VERSION} jacoco:cover into the makefile

ajnavarro commented on .travis.yml:
```diff
@@ -67,10 +67,12 @@ before_script:
-  - ./sbt test:scalastyle
+  - make scalastyle
+  - make test-scalastyle
   - ./sbt ++$TRAVIS_SCALA_VERSION jacoco:cover
```
ahaha, leave as it was is not the best solution for that problem. You should change this to: ./sbt ++${TRAVIS_SCALA_VERSION} jacoco:cover into the makefile

mcarmonaa replied:
Now it's fixed properly
